src/utils/simulation_results.jl

```diff
@@ -0,0 +1,210 @@
+        end
+    end
+    time_stamp[!, :Range] = convert(Array{Dates.DateTime}, time_stamp[!, :Range])
+    file_path = dirname(references[stage][variable[1]][1, :File_Path])
+    optimizer = read_json(joinpath(file_path, "optimizer_log.json"))
+    obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer["obj_value"])
+    if !isempty(dual)
+        duals = _read_references(duals, dual, stage, step, references, time_length)
+        results = DualResults(variables, obj_value, optimizer, time_stamp, duals)
+    else
+        results = SimulationResults(variables, obj_value, optimizer, time_stamp)
+    end
+    file_type = get(kwargs, :file_type, Feather)
+    write = get(kwargs, :write, false)
+    if write == true || :file_type in keys(kwargs)
+        write_results(
+            results,
+            SimulationResultsReference.results_folder,
+            "results";
+            file_type = file_type,
+        )
+    end
+    return results
+end
+
+"""
+    load_simulation_results(stage, SimulationResultsReference)
+
+This function goes through the reference table of file paths and
+aggregates the results over time into a struct of type OperationsProblemResults
+
+# Arguments
+- `SimulationResultsReference::SimulationResultsReference`: the container for the reference dictionary created in execute!
+- `stage_number::Int = 1`: The stage of the results getting parsed: 1 or 2
+
+# Example
+```julia
+stage = 2
+step = ["step-1", "step-2", "step-3"] # has to match the date range
+variable = [:P_ThermalStandard, :P_RenewableDispatch]
+results = load_simulation_results(stage, step, variable, SimulationResultsReference)
+```
+# Accepted Key Words
+- `write::Bool`: if true, the aggregated results get written back to the results file in the folder structure
+"""
+
+function load_simulation_results(
+    sim_output::SimulationResultsReference,
+    stage_name::String;
+    kwargs...,
+)
+    stage = "stage-$stage_name"
+    references = sim_output.ref
+    variables = Dict()
+    duals = Dict()
+    variable = (collect(keys(references[stage])))
+    dual = _find_duals(variable)
+    variable = setdiff(variable, dual)
+    time_stamp = DataFrames.DataFrame(Range = Dates.DateTime[])
+    time_length = sim_output.chronologies[stage]
+
+    for l in 1:length(variable)
+        date_df = references[stage][variable[l]]
+        variables[(variable[l])] = DataFrames.DataFrame()
+        for (ix, time) in enumerate(date_df.Date)
+            file_path = date_df[ix, :File_Path]
+            var = Feather.read(file_path)
+            variables[(variable[l])] = vcat(variables[(variable[l])], var[1:time_length, :])
+            if l == 1
+                time_stamp = vcat(time_stamp, _read_time(file_path, time_length))
+                check_file_integrity(dirname(file_path))
+            end
+        end
+    end
+    time_stamp[!, :Range] = convert(Array{Dates.DateTime}, time_stamp[!, :Range])
+    @show references[stage][variable[1]][1, :File_Path]
```

Comment:
This is failing. I'm not quite sure why, but the df in `references[stage][variable[1]]`  is empty. @Lilyhanig